test/functional/rpc_signrawtransactionwithkey.py

```diff
@@ -46,16 +49,12 @@
 
 class SignRawTransactionWithKeyTest(BitcoinTestFramework):
     def set_test_params(self):
-        self.setup_clean_chain = True
         self.num_nodes = 2
 
     def send_to_address(self, addr, amount):
-        input = {"txid": self.nodes[0].getblock(self.block_hash[self.blk_idx])["tx"][0], "vout": 0}
-        output = {addr: amount}
-        self.blk_idx += 1
-        rawtx = self.nodes[0].createrawtransaction([input], output)
-        txid = self.nodes[0].sendrawtransaction(self.nodes[0].signrawtransactionwithkey(rawtx, [self.nodes[0].get_deterministic_priv_key().key])["hex"], 0)
-        return txid
+        script_pub_key = address_to_scriptpubkey(addr)
+        tx = self.wallet.send_to(from_node=self.nodes[0],scriptPubKey=script_pub_key, amount=int(amount * COIN))
```

Comment:
nit: add spaces between passed arguments
 ```suggestion
         tx = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=script_pub_key, amount=int(amount * COIN))
```

Reply:
Why are you converting the amount to an integer here? Won’t that change the assumptions in the test? I see 49.999 explicitly passed in the test—don’t we want to maintain precision?

Reply:
I noticed that MiniWallet's `send_to` method doesn't play nice with non-integer amounts. It's failing because CTxOut's `nValue` is an int, so when you pass a float, the `to_bytes` method throws an error.

I did a quick scan of the codebase:

We've got 21 uses of MiniWallet's `send_to` method on the tests (not counting this PR)
- 17 of these are already sending an integer
- 4 are explicitly doing int(amount) before passing it to `send_to`

I think fixing this is probably outside what we're trying to do in this PR. I'm happy to create a separate issue for it and tackle it. 

@ismaelsadeeq, what do you think?

Reply:
@theStack what do you think?

Reply:
> It's failing because `CTxOut`'s `nValue` is an `int`, so when you pass a `float`, the `to_bytes` method throws an error.

I can also verify that this fails locally.

I'm generally not a fan of magic numbers, but the obvious explanation is that some satoshis were subtracted from the UTXO value to account for transaction fees.

If that's the case, then it's fine to round it down.

> I'm happy to create a separate issue for it and tackle it.

+1